envoy/network/connection.h

```diff
@@ -30,6 +30,7 @@ enum class ConnectionEvent {
   RemoteClose,
   LocalClose,
   Connected,
+  ConnectedZeroRtt,
```

Comment:
Instead of a different event, I think it would make more sense (from an API perspective) to add an accessor to query whether the connection was 0-rtt. I think that most consumers of connection events would want to treat `Connected` and `ConnectedZeroRtt` the same.

I'm also worried about code that says `if (event == Connected) else { /* this is definitely a close event */ }`, and introducting bugs by adding this. @mattklein123 wdyt?

Reply:
I think both events are needed on the client side. When "connected 0-RTT" event happens, the client can send early data, but needs to have the ability to retry the data in case the handshake fails and it can only send "safe" data (depending on the config). But when "Connected" happens, it can send everything. If we only have 1 event, how would a client distinguish between these two states?

Reply:
Got it, I understand now.

I just did a quick audit, and I don't see any problematic uses of the enum (code assuming only 3 enum values).

Reply:
I think there is still risk here, especially in private filters/extensions. I would recommend a release note about this. Given this can only be raised for H3 it's probably not risky enough for a runtime guard, but I think it's still worth calling out that people should audit their extensions?

Reply:
good point!